cmd/cluster-agent/api/agent/agent.go

```diff
@@ -126,15 +150,30 @@ func getCheckLatestEvents(w http.ResponseWriter, r *http.Request) {
 
 func getPodMetadata(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
+	var err error
+	var slcB []byte
 	nodeName := vars["nodeName"]
 	podName := vars["podName"]
-	svcList := as.GetPodServiceNames(nodeName, podName)
 
-	slcB, err := json.Marshal(svcList)
+	if podName == "*" {
+		log.Info("Computing service map on all nodes ...")
+		svcList, errNodes := as.GetServiceMapBundleOnNode(nodeName)
+		if errNodes != nil {
+			log.Errorf("could not collect the service map for %s: %s", nodeName, errNodes.Error())
+		}
+		slcB, err = json.Marshal(svcList)
```

Comment:
I think it's better to leave the part where we write the body after, as we deal with all the cases above.

Reply:
I think this controller need to be tested because you have a lot of conditions cascading on errors.
I don't think we should return something different than a json in case of 404.

This controller should be cut, fail fast and not carrying too much contexts between conditions to use an unique return point.

I feel like the error management inside maps is cumbersome and it's complex to keep a track of them. 
Because you are still returning a map fulfilled with the error and the error as error...
Maybe the aggregation of the errors should be done in the controller who needs to provide a json response.